### PR TITLE
Metadata. Also add schema to each Borsh value

### PR DESCRIPTION
**Pull request by nearmax**
* Adds `#[metadata]` macro, closing https://github.com/nearprotocol/NEPs/pull/3 for Rust API.
* Makes Borsh arguments, callbacks, and result also attach Borsh schema serialized in Borsh.

Blocked by https://github.com/nearprotocol/borsh/pull/57

### CODE REVIEW

File: near-bindgen-core/src/metadata_generator.rs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        #[no_mangle]
+        pub extern "C" fn metadata() {
+            let metadata = near_bindgen::Metadata::new(vec![
+                #(#methods),*
+            ]);
+            let data = borsh::try_to_vec_with_schema(&metadata).expect("Failed to serialize the metadata using Borsh");
+            near_bindgen::env::value_return(&data);
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (nearmax) [reply near-bindgen-core/src/metadata_generator.rs]: Done.

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,65 @@
+{
```
Comment (evgenykuzyakov) [inline .vscode/launch.json]: Remove?

File: Cargo.toml [hunk 3]
```diff
@@ -12,3 +12,9 @@ exclude = [
     "examples/status-message",
     "examples/status-message-collections",
 ]
+
+# Special triple # comment for ci.
+[patch.crates-io]
+### borsh = { git = "https://github.com/nearprotocol/borsh", branch = "master" }
+near-vm-logic = { git = "https://github.com/nearprotocol/nearcore", branch = "bump_borsh" }
```
Comment (evgenykuzyakov) [inline Cargo.toml]: Don't forget to remove

File: .vscode/launch.json (hunk 2 above)
Comment (nearmax) [reply .vscode/launch.json]: Done

File: Cargo.toml (hunk 3 above)
Comment (nearmax) [reply Cargo.toml]: Done

**Comment by evgenykuzyakov** [comment]
How to use metadata? @nearmax

**Pull request merged**